Recreate Quttera branch

hayzer opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=21350) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=21350) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=21350) it.</sub>

YuvHayun commented:
Hi @hayzer, thank you for your contribution.
Please fill the contribution form and let me know when the pr is ready for review.

YuvHayun commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner_test.py:
```diff
@@ -0,0 +1,61 @@
+import json
```
Please add a "given - when - then" documentation.
also, note that you have a few tests that fail.

YuvHayun commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py:
```diff
@@ -0,0 +1,179 @@
+    demisto.debug(f"Command called {command}")
+    demisto.results(f"Args are {args}")
+    demisto.results(f"params are {params}")
+    client = Client(params)
```
Let's also add a test-module command, 
test-module should test a simple argument less request - just to ensure there's a connection.

hayzer replied:
Fixed

YuvHayun reviewed (changes requested):
Hey @hayzer, please see my comments below.
Also, please make sure to fill the contribution form.

YuvHayun commented:
Hi @hayzer, we are now entering the holidays period and therefore we might not be available.
Sorry in advance for any long-delay that'll take us to reply.

edik24 commented:
Hi @hayzer,
any update on the requested changes?

hayzer commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/quttera-api.py:
```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python
```
Removed that file as it only serve the local test

hayzer commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/code-workspace.code-workspace:
```diff
@@ -0,0 +1,8 @@
+{
```
Removed as it's not relevant.

YuvHayun replied:
Seems like it wasn't removed.

hayzer commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml:
```diff
@@ -0,0 +1,156 @@
+script:
+  type: python
+  subtype: python3
+  dockerimage: demisto/python3:3.10.4.28442
```
Fixed

hayzer commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py:
```diff
@@ -0,0 +1,179 @@
+    :return: ``CommandResults`` object contain the start scan response
+    :rtype: ``CommandResults``
+    """
+    domain = args['domain']
```
Fixed

hayzer commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner_description.md:
```diff
@@ -0,0 +1,7 @@
+
+# Credentials
+
+Use API key supply by Quttera
```
Fixed

YuvHayun commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py:
```diff
@@ -0,0 +1,179 @@
+            "blacklist": f"/api/v3/{self.apikey}/blacklist/status"
+        }
+
+        op = operations[operation]
```
still need change

YuvHayun commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py:
```diff
@@ -0,0 +1,179 @@
+    """main function, parses params and runs command functions
+    """
+    demisto.debug(f"Command called {command}")
+    demisto.results(f"Args are {args}")
+    demisto.results(f"params are {params}")
```
Still need change.

YuvHayun reviewed (changes requested):
Hey @hayzer, please notice that are few unresolved comments that require your attention.
Also, please pull changes from master to update your repo, and make sure to fill out the contribution form.
Fill free to contact me if you have any further questions.

hayzer commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/README.md:
```diff
@@ -0,0 +1,22 @@
+Quttera API inspects URLs, domains or websites for security issues.
```
I've rerun this command multiple time but it has no effect on the README file.

hayzer commented on Packs/QutteraWebsiteMalwareScanner/pack_metadata.json:
```diff
@@ -0,0 +1,36 @@
+    "githubUser": [
+        "hayzer"
+    ]
+}
```
Fixed

YuvHayun commented:
@hayzer let me know when you're ready for re-review.

YuvHayun commented:
ֿHey @hayzer, still waiting for your comment & fixes.

hayzer commented:
Hi Yuval,

Yes, I know. I was overwhelmed with other tasks. Next week expects to fix
your comments.

Best regards,
Thomas


On Tue, Nov 22, 2022 at 1:13 PM Yuval Hayun ***@***.***>
wrote:

> ֿHey @hayzer <https://github.com/hayzer>, still waiting for your comment
> & fixes.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/demisto/content/pull/21350#issuecomment-1323570798>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAATVLJM46KY2ZQAI523R53WJS2HXANCNFSM6AAAAAAQQMJ4ZM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

YuvHayun commented:
Hi @hayzer, I see you started working on the pr.
Please let me know when it's ready to re-review.

hayzer commented:
Hi Yuval,

Indeed, we're making progress at the moment. I will let you know when we're
done.

Best regards,
Thomas


On Mon, Dec 5, 2022 at 8:27 AM Yuval Hayun ***@***.***> wrote:

> Hi @hayzer <https://github.com/hayzer>, I see you started working on the
> pr.
> Please let me know when it's ready to re-review.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/demisto/content/pull/21350#issuecomment-1336858842>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAATVLLYBSYJMPSQHJAW5HLWLWKM3ANCNFSM6AAAAAAQQMJ4ZM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

YuvHayun commented:
Hey @hayzer still waiting for your update.

hayzer commented:
Hi Yuval,

Could you help me with my question first? It seems it's the last thing
before the build is green.

Best regards,
Thomas


On Thu, Dec 22, 2022 at 9:29 AM Yuval Hayun ***@***.***>
wrote:

> Hey @hayzer <https://github.com/hayzer> still waiting for your update.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/demisto/content/pull/21350#issuecomment-1362551323>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAATVLLRSXCWISLADZV3KILWOQGN3ANCNFSM6AAAAAAQQMJ4ZM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

YuvHayun commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py:
```diff
@@ -0,0 +1,205 @@
+
+if __name__ in ('__main__', '__builtin__', 'builtins'):
+    try:
+        main(demisto.params(), demisto.args(), demisto.command())
```
```suggestion
        main()
```

YuvHayun commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py:
```diff
@@ -0,0 +1,205 @@
+    raise DemistoException(f'Service is down: ${resp}')
+
+
+def main(params: dict, args: dict, command: str):
+    """main function, parses params and runs command functions
+    """
+    demisto.info(f"Command called {command}")
```
```suggestion
def main()  # pragma: no cover
    """main function, parses params and runs command functions
    """
    params = demisto.params()
    args = demisto.args()
    command = demisto.command()
    demisto.info(f"Command called {command}")
```

YuvHayun reviewed (changes requested):
@hayzer I added two fixes suggestions, this should help you pass the lint step.

YuvHayun commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py:
```diff
@@ -0,0 +1,208 @@
+    raise DemistoException(f'Service is down: ${resp}')
+
+
+def main()  # pragma: no cover
```
```suggestion
def main():  # pragma: no cover
```

My bad, missed the ":"

YuvHayun commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml:
```diff
@@ -0,0 +1,156 @@
+script:
+  type: python
+  subtype: python3
+  dockerimage: demisto/python3:3.10.8.39276
```
```suggestion
  dockerimage: demisto/python3:3.10.9.40422
```

YuvHayun commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.py:
```diff
@@ -0,0 +1,208 @@
+    raise DemistoException(f'Service is down: ${resp}')
+
+
+def main():  # pragma: no cover
+    """main function, parses params and runs command functions
+    """
+    params = demisto.params()
+    args = demisto.args()
+    command = demisto.command()
+    demisto.info(f"Command called {command}")
+    demisto.info(f"Args are {args}")
+    demisto.info(f"params are {params}")
+    client = Client(params)
+    if command == 'quttera-scan-start':
+        results = scan_start(client, args)
+    elif command == 'quttera-rescan-status':
+        results = rescan_status(client, args)
+    elif command == 'quttera-scan-report':
+        results = scan_report(client, args)
+    elif command == 'quttera-report-malware':
+        results = report_malware(client, args)
+    elif command == 'quttera-status-blacklist':
+        results = status_blacklist(client, args)
+    elif command == 'test-module':
+        results = test_module(client)
+    else:
+        raise NotImplementedError(f'Command {command} not implemented')
+    return_results(results)
+
+
+if __name__ in ('__main__', '__builtin__', 'builtins'):
+    try:
+        main()
+    except Exception as exception:
+        return_error(exception)
```
```suggestion
def main():  # pragma: no cover
    """main function, parses params and runs command functions
    """
    params = demisto.params()
    args = demisto.args()
    command = demisto.command()
    demisto.info(f"Command called {command}")
    demisto.info(f"Args are {args}")
    demisto.info(f"params are {params}")
    client = Client(params)
    try:
      if command == 'quttera-scan-start':
          results = scan_start(client, args)
      elif command == 'quttera-rescan-status':
          results = rescan_status(client, args)
      elif command == 'quttera-scan-report':
          results = scan_report(client, args)
      elif command == 'quttera-report-malware':
          results = report_malware(client, args)
      elif command == 'quttera-status-blacklist':
          results = status_blacklist(client, args)
      elif command == 'test-module':
          results = test_module(client)
      else:
            raise ValueError(f'Command {command} is not implemented in this integration')
      return_results(results)
    except Exception as e:
        return_error(f'Failed to execute {command} command.\nError:\n{str(e)}')


if __name__ in ('__main__', '__builtin__', 'builtins'):
        main()
```

let's put the try and except inside the main only for the party where we ran the commands.

YuvHayun commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/README.md:
```diff
@@ -0,0 +1,140 @@
+Quttera Website Malware Scanner
```
Just making sure, did you run the generate-docs command after all your fixes? the readme file is up to date?

YuvHayun reviewed (changes requested):
Hey @hayzer done another review after your fixes, job was very good. Left some minor comments for you to fix.
After you'll fix my comments I'll trigger the workflow tests and when those are passed, we'll schedule a short demo with @edik24

YuvHayun commented:
Hey @hayzer 
I pushed some minor fixes myself to fix indentation issues.
I ran instance tests and everything passed.

The next step is a demo, @edik24 will send you an email to schedule a 30 minutes demo for next week.
Please make sure to fill out the contribution form before the demo.
Also regarding my [question](https://github.com/demisto/content/pull/21350#discussion_r1056955402) please re-run generate-docs if you haven't done that yet.

hayzer commented:
Hi. I'm traveling. Will be offline till tomorrow. We'll try to create short
video asap.

On Wed, Dec 28, 2022, 10:04 Yuval Hayun ***@***.***> wrote:

> Hey @hayzer <https://github.com/hayzer>
> I pushed some minor fixes myself to fix indentation issues.
> I ran instance tests and everything passed.
>
> The next step is a demo, @edik24 <https://github.com/edik24> will send
> you an email to schedule a 30 minutes demo for next week.
> Please make sure to fill out the contribution form before the demo.
> Also regarding my question
> <https://github.com/demisto/content/pull/21350#discussion_r1056955402>
> please re-run generate-docs if you haven't done that yet.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/demisto/content/pull/21350#issuecomment-1366485789>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAATVLJCN75KO7RPTKHG6UTWPP7A5ANCNFSM6AAAAAAQQMJ4ZM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

YuvHayun commented:
Hey @hayzer, following our demo - few comments to fix:
- Let's add the default URL as the default value of the URL integration param.
- Let's add a comment to tell the user to use the status command after executing the scan command if he wants to get the current execution status.
- Let's edit the human-readable output, no need to add the status code to the human-readable, for commands like the blacklist, we can return free text / a table with three columns - the name, the type (URL, domain, etc..) and blacklisted yes/no.
The integration which we gave as an example in the demo is Microsoft Defender Advanced Threat Protection.
In your attached video please show new readable output after all the mentioned fixes. thx again!

YuvHayun commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/README.md:
```diff
@@ -0,0 +1,140 @@
+Quttera Website Malware Scanner
+This integration was integrated and tested with version xx of QutteraWebsiteMalwareScanner
```
Replace xx with the tested API version.

hayzer replied:
Fixed

YuvHayun commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml:
```diff
@@ -0,0 +1,157 @@
+script:
+  type: python
+  subtype: python3
+  dockerimage: demisto/python3:3.10.9.42008
```
```suggestion
  dockerimage: demisto/python3:3.10.9.44472
```

hayzer replied:
Solved

YuvHayun commented on Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml:
```diff
@@ -144,7 +145,7 @@ script:
       description: The name of the domain to retrieve its blacklist and reputation
     outputs:
-    - contextPath: QutteraWebsiteMalwareScanning.domain
+    - contextPath: QutteraWebsiteMalwareScanning.report
```
why did you change this one?

hayzer replied:
Solved

YuvHayun reviewed (changes requested):
Hey @hayzer I added a few comments to fix.
Also:
1. Please note your unit tests failed. 
2. Regarding the video - let's make sure to include commands where you changed the readable output (where we added more informational msgs and tables), and also show the cases where the execution fails (like in status_blacklist where you're expecting 404) and context output 

Let me know when you're ready for rereview.

hayzer commented:
Hi Yuval,

Can you take a look, please? I think I fixed most of the things.

Best regards,
Thomas


On Sun, Jan 22, 2023 at 11:41 AM Yuval Hayun ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Hey @hayzer <https://github.com/hayzer> I added a few comments to fix.
> Also:
>
>    1. Please note your unit tests failed.
>    2. Regarding the video - let's make sure to include commands where you
>    changed the readable output (where we added more informational msgs and
>    tables), and also show the cases where the execution fails (like in
>    status_blacklist where you're expecting 404) and context output
>
> Let me know when you're ready for rereview.
> ------------------------------
>
> In
> Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/README.md
> <https://github.com/demisto/content/pull/21350#discussion_r1083427731>:
>
> > @@ -0,0 +1,140 @@
>
> +Quttera Website Malware Scanner
>
> +This integration was integrated and tested with version xx of QutteraWebsiteMalwareScanner
>
>
> Replace xx with the tested API version.
> ------------------------------
>
> In
> Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml
> <https://github.com/demisto/content/pull/21350#discussion_r1083427835>:
>
> > +  scan of a domain or URL,
>
> +  retrieve malware scan report and blacklisting information.
>
> +configuration:
>
> +- display: API Key
>
> +  name: apikey
>
> +  required: true
>
> +  type: 4
>
> +- display: Quttera Scanner URL
>
> +  name: base_url
>
> +  required: true
>
> +  defaultvalue: "https://scannerapi.quttera.com"
>
> +  type: 0
>
> +script:
>
> +  type: python
>
> +  subtype: python3
>
> +  dockerimage: demisto/python3:3.10.9.42008
>
>
> ⬇️ Suggested change
>
> -  dockerimage: demisto/python3:3.10.9.42008
>
> +  dockerimage: demisto/python3:3.10.9.44472
>
>
> ------------------------------
>
> In
> Packs/QutteraWebsiteMalwareScanner/Integrations/QutteraWebsiteMalwareScanner/QutteraWebsiteMalwareScanner.yml
> <https://github.com/demisto/content/pull/21350#discussion_r1083427961>:
>
> > @@ -144,7 +145,7 @@ script:
>
>        secret: false
>
>        description: The name of the domain to retrieve its blacklist and reputation
>
>      outputs:
>
> -    - contextPath: QutteraWebsiteMalwareScanning.domain
>
> +    - contextPath: QutteraWebsiteMalwareScanning.report
>
>
> why did you change this one?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/demisto/content/pull/21350#pullrequestreview-1264746358>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAATVLICGY5ID2E6LP7E6YTWTUFEBANCNFSM6AAAAAAQQMJ4ZM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

YuvHayun commented:
Hey @hayzer, the fixes looks great! the only thing that is missing now is the video.
Let's make sure to include commands where you changed the readable output (where we added more informational msgs and tables), and also show the cases where the execution fails (like in status_blacklist where you're expecting 404) and context output.
Please try to record a quick video asap and I'll approve and merge the issue.

YuvHayun commented:
@hayzer I got your video sent via mail.
looks great!
merging for internal PR where i'll also fix the last validation.
You should be able to see the pack in the marketplace tomorrow evening.
Thx a lot for your contribution!
